ImageInLib/ImageInLib/src/imageInterpolation.c

```diff
@@ -3,257 +3,145 @@
+        return neighbors.top_left;
+    }
+    if (min_indice == 1) {
+        return neighbors.top_right;
+    }
+    if (min_indice == 2) {
+        return neighbors.bottom_left;
+    }
+    if (min_indice == 3) {
+        return neighbors.bottom_right;
+    }
 }
 
+bool interpolateImageFromImageCStoRealWorldCS(Image_Data2D src_image, Image_Data2D interp_image, dataType scale_factor) {
 
-//Get 3D point in image coordinates system from 3D point in real coordinate system
-//TO DO : find manually the matrix inverse in order to write the final function 
-Point3D realCoordToImageCoord(Point3D srcPoint, Point3D realOrigin, VoxelSpacing imageSpacing, OrientationMatrix orientation) {
-    Point3D resultPoint;
-
-    return resultPoint;
-}
-
-/*
-* This function perform 2d nearest neighbor interpolation (expansion or shrinking)
-*/
-bool resizeImage(Image_Data2D oldImage, Image_Data2D newImage) {
+    //The structure interp_image represents the image in real world coordinates system
+    //the spacing is the real spacing
+    //the origin is the real origin
+    //the orientation is the real orientation
 
-    if (oldImage.imageDataPtr == NULL || newImage.imageDataPtr == NULL)
+    if (src_image.imageDataPtr == NULL || interp_image.imageDataPtr == NULL)
         return false;
 
-    const size_t height_old = oldImage.height, width_old = oldImage.width;
-    const size_t height_new = newImage.height, width_new = newImage.width;
+    const size_t src_height = src_image.height, src_width = src_image.width;
+    const size_t interp_height = interp_image.height, interp_width = interp_image.width;
 
     int i, j;
-    dataType i_new, j_new;
+    Point2D src_image_point, interp_image_point, src_rw_point, interp_rw_point;
 
-    int i_int, j_int;
+    for (i = 0; i < interp_height; i++) {
+        for (j = 0; j < interp_width; j++) {
 
-    int i_floor, i_ceil;
-    int j_floor, j_ceil;
+            //Interpolated image point
+            interp_image_point.x = i;
+            interp_image_point.y = j;
 
-    //Compute scale factor
-    dataType sx = (dataType)height_old / height_new;
-    dataType sy = (dataType)width_old / width_new;
+            //Get the source point correponding to interpolated point
+            src_image_point.x = interp_image_point.x * (1. / scale_factor);
+            src_image_point.y = interp_image_point.y * (1. / scale_factor);
 
-    dataType val = 0.0;
+            //Get src point in real world CS
+            src_rw_point = getRealCoordFromImageCoord2D(src_image_point, interp_image.origin, interp_image.spacing, interp_image.orientation);
 
-    //Map back to original coordinates
-    for (i = 0; i < height_new; i++) {
-        for (j = 0; j < width_new; j++) {
+            //Get the nearest neighbor in real world coordinates system
+            //The nearest neighbor is found with real world cordinate system
+            Point2D nearest_neighbor = getNearestNeighbor2D(src_rw_point, interp_image.spacing);
```

Comment:
in my opinion, this is not correct idea.
You expect standard orthogonal coord. system again...

In my function you should base the finding of neighbors on image coordinates of src image (you can expect orhonormal CS in such case)
       n
       ||
w = p = e
       ||
       s

and then you can consider pixel spacing for particular interpolation function, which would return just interpolated value (not coordinates)

Reply:
here I did the transformation before, so for the function I am presenting, the neighbors are found in real world coordinates system. Every point x, in the coordinates system can be written x = a * i + b * j. so when I divide a by the spacing sx I get approximative value of a and the by doing floor and ceil, I found the neighbors of the current point

Reply:
We do not need to fins neighbors in real world coordinates.
Image coordinate systems are orthonormal (0,1)x(1,0)
If we know pixel spacing, we can say, which pixel center is the closest to particular 2d point in image CS